File: java/eyes.sdk.core/src/main/java/com/applitools/eyes/config/IConfiguration.java
package com.applitools.eyes.config;

import com.applitools.eyes.*;

import java.net.URI;

public interface IConfiguration {
    Boolean getSaveNewTests();

    Configuration setSaveNewTests(Boolean saveNewTests);

    Boolean getSaveFailedTests();

    Configuration setSaveFailedTests(Boolean saveFailedTests);

    ImageMatchSettings getDefaultMatchSettings();

    Configuration setDefaultMatchSettings(ImageMatchSettings defaultMatchSettings);

    Integer getMatchTimeout();

    Configuration setMatchTimeout(Integer matchTimeout);

    String getHostApp();

    Configuration setHostApp(String hostApp);

    String getHostOS();

    Configuration setHostOS(String hostOS);

    Integer getStitchOverlap();

    StitchOverlap getOverlap();

    Configuration setStitchOverlap(Integer stitchingOverlap);

    Configuration setStitchOverlap(StitchOverlap stitchOverlap);

    BatchInfo getBatch();

    Configuration setBatch(BatchInfo batch);

    String getBranchName();

    Configuration setBranchName(String branchName);

    String getAgentId();

    Configuration setAgentId(String agentId);

    String getParentBranchName();

    Configuration setParentBranchName(String parentBranchName);

    String getBaselineBranchName();

    Configuration setBaselineBranchName(String baselineBranchName);

    String getBaselineEnvName();

    String getEnvironmentName();

    Configuration setEnvironmentName(String environmentName);

    Configuration setBaselineEnvName(String baselineEnvName);

    Boolean getSaveDiffs();

    Configuration setSaveDiffs(Boolean saveDiffs);

    String getAppName();

    Configuration setAppName(String appName);

    String getTestName();

    Configuration setTestName(String testName);

    RectangleSize getViewportSize();

    Configuration setViewportSize(RectangleSize viewportSize);

    SessionType getSessionType();

    Configuration setSessionType(SessionType sessionType);

    @Deprecated
    FailureReports getFailureReports();

    @Deprecated
    Configuration setFailureReports(FailureReports failureReports);

    Boolean isSendDom();

    Configuration setSendDom(Boolean sendDom);

    Boolean getIgnoreCaret();

    Configuration setIgnoreCaret(Boolean value);

    String getApiKey();

    Configuration setApiKey(String apiKey);

    URI getServerUrl();

    Configuration setServerUrl(String serverUrl);

    AbstractProxySettings getProxy();

    Configuration setProxy(AbstractProxySettings proxy);

    MatchLevel getMatchLevel();

    Configuration setMatchLevel(MatchLevel matchLevel);

    Boolean getIgnoreDisplacements();

    Configuration setIgnoreDisplacements(Boolean isIgnoreDisplacements);

    AccessibilitySettings getAccessibilityValidation();

    Configuration setAccessibilityValidation(AccessibilitySettings accessibilitySettings);

    Boolean getUseDom();

    Configuration setUseDom(Boolean useDom);

    Boolean getEnablePatterns();

    Configuration setEnablePatterns(Boolean enablePatterns);
}

File: dotnet/Eyes.Images.DotNet/IConfiguration.cs
using System;
using System.Collections.Generic;
using Applitools.Commands;
using Applitools.Utils.Geometry;
using Applitools.VisualGrid;

namespace Applitools
{
    public interface IConfiguration
    {
        string AgentId { get; set; }

        string AppName { get; set; }

        string BaselineBranchName { get; set; }

        string BaselineEnvName { get; set; }

        BatchInfo Batch { get; set; }

        string BranchName { get; set; }

        ImageMatchSettings DefaultMatchSettings { get; set; }

        string EnvironmentName { get; set; }

        string HostApp { get; set; }

        string HostOS { get; set; }

        bool? IgnoreCaret { get; set; }

        TimeSpan? MatchTimeout { get; set; }

        string ParentBranchName { get; set; }

        bool? SaveDiffs { get; set; }

        bool? SaveNewTests { get; set; }
        bool? SaveFailedTests { get; set; }

        bool? SendDom { get; set; }

        int? StitchOverlap { get; set; }

        string TestName { get; set; }

        RectangleSize ViewportSize { get; set; }

        MatchLevel? MatchLevel { get; set; }

        string ServerUrl { get; set; }

        string ApiKey { get; set; }

        bool? IgnoreDisplacements { get; set; }

        bool? UseDom { get; set; }

        bool? EnablePatterns { get; set; }

        AccessibilitySettings AccessibilityValidation { get; set; }
        int? AbortIdleTestTimeout { get; set; }

        bool? LayoutBreakpointsEnabled { get; set; }
        
        LayoutBreakpointsOptions LayoutBreakpoints { get; set; }
        
        ProxySettings Proxy { get; set; }
        bool? IsForceFullPageScreenshot { get; set; }

        int? WaitBeforeScreenshots { get; set; }

        StitchModes? StitchMode { get; set; }

        bool? HideScrollbars { get; set; }

        bool? HideCaret { get; set; }

        bool? DisableBrowserFetching { get; set; }
        IImageCrop Cut { get; set; }


        VisualGridOption[] VisualGridOptions { get; set; }

        bool? UseCookies { get; set; }
        bool? DontCloseBatches { get; set; }
        DebugScreenshotHandler DebugScreenshotProvider { get; set; }


        List<RenderBrowserInfo> GetBrowsersInfo();


        IConfiguration AddBrowsers(params IRenderBrowserInfo[] browsersInfo);
        IConfiguration AddBrowser(DesktopBrowserInfo browserInfo);
        IConfiguration AddBrowser(ChromeEmulationInfo emulationInfo);
        IConfiguration AddBrowser(IosDeviceInfo deviceInfo);
        IConfiguration AddBrowser(AndroidDeviceInfo deviceInfo);
        IConfiguration AddBrowser(int width, int height, BrowserType browserType, string baselineEnvName = null);

        IConfiguration AddDeviceEmulation(DeviceName deviceName,
            ScreenOrientation screenOrientation = ScreenOrientation.Portrait,
            string baselineEnvName = null);

        IConfiguration SetWaitBeforeScreenshots(int? waitBeforeScreenshots);

        IConfiguration SetStitchMode(StitchModes? stitchMode);

        IConfiguration SetHideScrollbars(bool? hideScreenshot);

        IConfiguration SetHideCaret(bool? hideCaret);

        IConfiguration SetDisableBrowserFetching(bool? disableBrowserFetching);

        IConfiguration SetForceFullPageScreenshot(bool? forceFullPageScreenshot);

        IConfiguration SetAgentId(string value);
        IConfiguration SetAppName(string value);
        IConfiguration SetBaselineBranchName(string value);
        IConfiguration SetBaselineEnvName(string value);
        IConfiguration SetBatch(BatchInfo value);
        IConfiguration SetBranchName(string value);
        IConfiguration SetDefaultMatchSettings(ImageMatchSettings value);
        IConfiguration SetEnvironmentName(string value);
        IConfiguration SetHostApp(string value);
        IConfiguration SetHostOS(string value);
        IConfiguration SetIgnoreCaret(bool? value);
        IConfiguration SetMatchTimeout(TimeSpan? value);
        IConfiguration SetParentBranchName(string value);
        IConfiguration SetSaveDiffs(bool? value);
        IConfiguration SetSaveNewTests(bool? value);
        IConfiguration SetSaveFailedTests(bool? value);
        IConfiguration SetSendDom(bool? value);
        IConfiguration SetStitchOverlap(int? value);
        IConfiguration SetTestName(string value);
        IConfiguration SetViewportSize(RectangleSize value);
        IConfiguration SetMatchLevel(MatchLevel? value);
        IConfiguration SetServerUrl(string value);
        IConfiguration SetApiKey(string value);
        IConfiguration SetDontCloseBatches(bool? value);
        IConfiguration SetDebugScreenshotProvider(DebugScreenshotHandler value);
        IConfiguration SetCut(IImageCrop value);

        IConfiguration SetProxy(ProxySettings value);
        IConfiguration SetIgnoreDisplacements(bool? value);
        IConfiguration SetUseDom(bool? value);
        IConfiguration SetEnablePatterns(bool? value);
        IConfiguration SetAccessibilityValidation(AccessibilitySettings value);
        IConfiguration SetAbortIdleTestTimeout(int? value);
        IConfiguration SetLayoutBreakpointsEnabled(bool? shouldSet);
        IConfiguration SetLayoutBreakpoints(params int[] breakpoints);
        IConfiguration SetLayoutBreakpoints(bool layoutBreakpointsEnabled);
        IConfiguration SetLayoutBreakpoints(LayoutBreakpointsOptions options);
        IConfiguration SetVisualGridOptions(params VisualGridOption[] options);

        IConfiguration SetUseCookies(bool? useCookies);
    }
}